prometheusbp/promtest/testutil.go

```diff
@@ -40,6 +40,50 @@ func (p *PrometheusMetricTest) CheckSampleCountDelta(delta int) {
 	}
 }
 
+// CheckDeltaLE checks that the metric value changes less or equal to delta from when Helper was called.
+func (p *PrometheusMetricTest) CheckDeltaLE(delta float64) {
+	p.tb.Helper()
+
+	got, _ := p.getValueAndSampleCount()
+	got -= float64(p.initValue)
```

Comment:
also if we really need all the new APIs then this part should be extracted to its own helper function as they are repeated several times now.